### PR TITLE
The gimbal mixin maintains semantic property of flex mixin attributes

### PR DESCRIPTION
**Pull request by paxperscientiam**
So, how to explain this?

As indicated in issue  #9837, the new and handy flex mixins -- though helpful -- become difficult to use with, for instance, `flex-direction: column`. Because the mixin argument-value pairs for `@mixin flex-align(...)` are biased in favor of `flex-direction: row`, the semantic meaning is lost/confused with `flex-direction: column`.

I put together a rough cut of what I am calling the "gimbal" mixin. It allows for the 'top', 'bottom', 'left', 'right', etc attribute values to retain their semantic value irrespective of flex direction.

This is just a draft and I would like some input on how "justify" and "spaced" should relate to "stretch". [linky](http://foundation.zurb.com/sites/docs/flexbox.html#flex-align).

EDIT: fixed incorrect reference

### CODE REVIEW

File: scss/components/_flex.scss [hunk 1]
```diff
@@ -32,3 +32,30 @@
...
+@mixin gimbal($o:row,$x:left,$y:top) {
+  //columns
+  $map-xy: (
+  right: bottom,
+  left: top,
+  top: left,
+  bottom: right,
+  center: middle,
+  middle: center
+);
```
Comment (IamManchanda) [inline scss/components/_flex.scss]: Indentation !!!

File: scss/components/_flex.scss [hunk 2]
```diff
@@ -32,3 +32,30 @@
     }
   }
 }
+
+@mixin gimbal($o:row,$x:left,$y:top) {
```
Comment (IamManchanda) [inline scss/components/_flex.scss]: Why the word "Gimbal" ?

File: scss/components/_flex.scss [hunk 3]
```diff
@@ -32,3 +32,30 @@
...
+
+  @include flex;
+  @include flex-direction($o);
+  @include flex-align($x:$x1,$y:$y1);
+}
+
+///////
+.gimbal-test {
+  @include gimbal($o:column,$x:left,$y:top);
+}
```
Comment (IamManchanda) [inline scss/components/_flex.scss]: Why this is here ?
No example css/scss please within here ?

--------------------------------------------------------------------------------------------------------

Although yes, this arise a question on how do you push a testing sass mixins

**Review by IamManchanda — COMMENTED** [review]
Please look into it!

File: scss/components/_flex.scss (hunk 2 above)
Comment (paxperscientiam) [inline scss/components/_flex.scss]: "Orientation" of natural directions remains constant regardless of whether or not the flex container flexes row or flexes column :)
[real world example](https://www.youtube.com/watch?v=29qeHVT-8-I)

File: scss/components/_flex.scss (hunk 1 above)
Comment (paxperscientiam) [inline scss/components/_flex.scss]: Oh, I know. I will fix this at the end -- after all the technical stuff is sorted out.

File: scss/components/_flex.scss (hunk 3 above)
Comment (paxperscientiam) [inline scss/components/_flex.scss]: That will definitely be deleted! Just there for testing purposes.

> ...example css/scss please...

Good idea, I will add that.

File: scss/components/_flex.scss (hunk 2 above)
Comment (IamManchanda) [inline scss/components/_flex.scss]: Perfect then ....

Comment (paxperscientiam) [inline scss/components/_flex.scss]: Possibly better to call it `flex-gimbal`.

Comment (IamManchanda) [inline scss/components/_flex.scss]: Yeah `flex-gimbal` will be better

**Comment by paxperscientiam** [comment]
OK, so I made the recommended changes. With regard to other flex-align options, still thinking.

File: scss/components/_flex.scss [hunk 4]
```diff
@@ -27,8 +27,52 @@
...
+//   justify-content: flex-start;
+//   align-items: flex-start;
+// }
+//  @include flex-gimbal($o:row,$x:right,$y:bottom) yields...
+// .gimbal-test {
+//   display: flex;
+//   flex-direction: row;
+//   justify-content: flex-end;
+//   align-items: flex-end;
+// }
```
Comment (IamManchanda) [inline scss/components/_flex.scss]: Please remove these comments

**Review by IamManchanda — CHANGES_REQUESTED** [review]

File: _vendor/normalize-scss/sass/normalize/_normalize-mixin.scss [hunk 5]
```diff
@@ -424,7 +424,7 @@
     optgroup,
     select,
     textarea {
-      font-family: inherit; /* 1 */
+      font-family: $base-font-family; /* 1 */
```
Comment (DaSchTour) [inline _vendor/normalize-scss/sass/normalize/_normalize-mixin.scss]: I guess this doesn't belong here. Looks like another topic and should be discussed in a separate thread.

**Comment by paxperscientiam** [comment]
This is off topic @IamManchanda , but do you know if anyone is paying attention to foundation-cli?

I submitted a PR with minor fixes, but I am interested in submitting more.

**Comment by IamManchanda** [comment]
@paxperscientiam it [happens](https://github.com/zurb/foundation-zurb-template/pull/53) bro

**Comment by kball** [comment]
@rafibomb can you take a look at this?

File: scss/components/_flex.scss [hunk 6]
```diff
@@ -32,3 +32,41 @@
...
+  @include flex-gimbal($o: column, $x: justify, $y: middle);
+}
+
+// @include flex-gimbal($o:column,$x:left,$y:top) yields...
+// .gimbal-test {
+//   display: flex;
+//   flex-direction: column;
+//   justify-content: flex-start;
+//   align-items: flex-start;
+// }
```
Comment (IamManchanda) [inline scss/components/_flex.scss]: Please Remove these comments chris

Comment (paxperscientiam) [inline scss/components/_flex.scss]: @IamManchanda sorry. done.

**Comment by kball** [comment]
@paxperscientiam can you write up some info on what exactly this is and how to use it, probably on the flexbox doc page under helpers?

**Comment by DanielRuf** [comment]
Some info about this in the docs might be very helpful.

@paxperscientiam do you need any assistance with this?

**Comment by paxperscientiam** [comment]
@DanielRuf I would/could if I had the time. Would you like the PR? I don't know if there's an official way, but I could close and would have no problem with you copying the code.

**Comment by DanielRuf** [comment]
We can push changes to your branch.
Let's check together what we need in the docs.

How could this be best described for the docs or written?

**Comment by paxperscientiam** [comment]
> How could this be best described for the docs or written?

I'd say something along the lines that `@mixin flex-gimbal(orientation, x, y)` frees devs from having to conceptualize a rotated coordinate system. That is, "x" (horizontal), means "x" irrespective if flex is row or column.

Does that make sense or help? Were you able to play with the code?

**Comment by kball** [comment]
I'm going to close this for now as it was never finished out and we're trying to consolidate up fixes and outstanding PRs for F6 now that ZURB is releasing more control to the community. We can reconsider for F7.

**Pull request closed without merging**